### PR TITLE
chore: Separate release please script

### PR DESCRIPTION
**Pull request by ddixit14**
This PR makes release-please config generation separate from bootstrap.sh
Also fixes the error where modules which did not have a bom get picked up for config generation. 
 
NOTE: This is just a modified version of release-please script which we need to use for manually staging the modules. Once release-please starts working totally fine, this will be changed again.

### CODE REVIEW

File: templates/generation/release_please_config_generation.sh [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+
+set -e
+
+echo "{" > .release-please-manifest.json
+
+echo ""\""google-cloud-gapic-bom"\"": "\""0.0.0"\""," >> .release-please-manifest.json
+
+GENERATION_DIR=$(dirname -- "$0");
+
+num_modules="$(wc -l < ../../repos.txt)"
```
Comment (suztomo) [inline templates/generation/release_please_config_generation.sh:11]: Don't rely on repos.txt

File: templates/generation/release_please_config_generation.sh [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+
+rp_config_line=""
+rp_manifest_line=""
+
+for bom_directory in $(find . -name 'google-*-bom' | sort); do
```
Comment (suztomo) [inline templates/generation/release_please_config_generation.sh:16]: Does this (Release Please config) really need to work based on "-bom" directory? To me, it just need to find versions.txt, regardless of "-bom" directories.

You have the second loop which does not rely on "-bom" directory.

File: templates/generation/release_please_config_generation.sh (hunk 1 above)
Comment (ddixit14) [reply templates/generation/release_please_config_generation.sh:11]: What should I rely on, then? As a source of truth to calculate the total number of modules in the config files?

Comment (suztomo) [reply templates/generation/release_please_config_generation.sh:11]: The number of the modules.

File: bootstrap.sh [hunk 3]
```diff
@@ -101,103 +101,7 @@ echo ""\""google-cloud-gapic-bom"\"": "\""0.0.0"\""," >> .release-please-manifes
...
-
-  #adding the line to manifest config file
-  echo "${rp_manifest_line}" >> .release-please-manifest.json
-done
-
-echo "}" >> .release-please-manifest.json
-
-awk -v "packagesList=$rp_config_line" '{gsub(/ALL_PACKAGES/,packagesList)}1' \
-    ../../release_please_config_raw.json > release-please-config.json
+bash generation/release_please_config_generation.sh
```
Comment (suztomo) [inline bootstrap.sh:104]: Can you rename the script as "generate_release_please_config.sh"?

I put the verb first in other scripts.

File: templates/generation/release_please_config_generation.sh (hunk 2 above)
Comment (ddixit14) [reply templates/generation/release_please_config_generation.sh:16]: This is not just generating release-please config. It is also managing versions as we want them. Sometimes it is according to release-please functioning, and right now it is according to upload to staging directory process. 
Also, there are different kind of modules (some having more modules than others), and hence their versions.txt are built differently. 
I'm writing the logic differently now. Earlier assumption that this loop covered all the modules proved to be incorrect, the main challenge here is making the versions sync.

**Comment by suztomo** [comment]
Can you figure out how to stop generating the backup file?

Screen Shot 2022-08-25 at 12 09 31 AM

File: templates/generation/release_please_config_generation.sh (hunk 2 above)
Comment (suztomo) [reply templates/generation/release_please_config_generation.sh:16]: > I'm writing the logic differently now. 

Good. Looking forward. Consider creating a loop by finding all "versions.txt" files.

**Comment by ddixit14** [comment]
> Can you figure out how to stop generating (or adding) the backup files?
> 
> Screen Shot 2022-08-25 at 12 09 31 AM

Fixed.

File: bootstrap.sh (hunk 3 above)
Comment (ddixit14) [reply bootstrap.sh]: Done.

File: templates/generation/release_please_config_generation.sh (hunk 2 above)
Comment (ddixit14) [reply templates/generation/release_please_config_generation.sh]: As discussed, this script will loop over the modules, generate release-please config files and sync the versions. 
However, there is one category of modules which might not get updated -> single component modules (like java-dns, java-notification, java-grafeas) with both current and released versions as non-snapshot in versions.txt. 
I can fix it later if it poses issues for us.

File: templates/generation/release_please_config_generation.sh (hunk 1 above)
Comment (ddixit14) [reply templates/generation/release_please_config_generation.sh]: Done.

**Review by suztomo — APPROVED** [review]

File: templates/generation/generate_release_please_config.sh [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+      do
+         if [[ ${line} =~ [0-9] ]] && ! [[ ${line} == *"SNAPSHOT"* ]] ; then
+           artifact_name=$(echo "${line}" | awk -F':' '{print $1}')
+           old_version=$(echo "${line}" | awk -F':' '{print $3}')
+           new_version=$(echo ${old_version} |  awk -F'.' '{print $1"."$2"."$3+1}' |  sed s/[.]$//)
+           new_version="${new_version}-SNAPSHOT"
+           sed -i "s|${artifact_name}:${old_version}:${old_version}|${artifact_name}:${old_version}:${new_version}|" ${version_file}
+           artifact_directory="${module}/${artifact_name}"
+           mvn -B -ntp -f ${artifact_directory} -U versions:set -DnewVersion=${new_version}
+           mvn versions:commit
```
Comment (suztomo) [inline templates/generation/generate_release_please_config.sh:65]: This runs against all projects, making the runtime unnecessary longer.

Comment (ddixit14) [reply templates/generation/generate_release_please_config.sh:65]: replaced it by adding " -DgenerateBackupPoms=false " in the above mvn versions:set command.

**Pull request merged**